src/components/JobCard/JobCard.tsx

```diff
@@ -40,7 +34,7 @@ function JobCard({
     } catch (e) {
       console.log(e);
     }
-    setPendingJobs(pendingJobs.filter((_, index) => index !== idx));
+    filterList(idx);
   }
 
   async function removeJob() {
```

Comment:
If I'm understanding correctly, this is your current design:

We have a callback in the parent (ApprovalScreen/FeedScreen) to remove the card from the list. In the child (JobCard), we hide the modal, trigger the remove firebase query, and then call that callback function to remove the card from the list.

I think splitting the 2 remove actions (query + remove from list) is really unintuitive, so I don't think we should split that logic -- instead, I would just lift all the removal-related logic into the callback from the parent. Then when you click **REMOVE**, it'd just call `setModalVisible(false)` and trigger your `onRemove` callback which contains all the removal logic.

Reply:
I found that it was necessary to split the logic as such in order to use `filteredApprovedJobs`/`notApprovedJobs` state hooks from within `JobCard` without passing the hooks in directly. Is there a different way this can be done?